File: codingame/community/JackSilverTheCasinoPart1/JackSilverTHeCasino1.scala
import math._
import scala.util._
object Solution extends App {
    val rnds = readInt
    var cash = readDouble
    for(i <- 0 until rnds) {
        val play = readLine
    
        var plays = play.split(" ");    
        var bet:Double = ceil(cash/4);
        cash = cash - bet;
        var ball:Int = plays(0).toInt;
        var call = plays(1);
        var target:Int = -1;
        if (call == "PLAIN" ) {
            target = plays(2).toInt;
        }   
        
        if (call == "PLAIN" && ball == target) {
            cash = cash + (bet * 36);
        }else if (call == "EVEN" && ball > 0 && ball % 2 == 0) {
            cash = cash + (bet * 2);
        }else if (call == "ODD" && ball % 2 == 1) {
            cash = cash + (bet * 2);
        }
    }
    println(cash.toInt)
}

File: codingame/community/JackSilverTheCasinoPart1/JackSilverTHeCasino1.js
var rounds = parseInt(readline());
var cash = parseInt(readline());
for (var i = 0; i < rounds; i++) {
    var play = readline().split(" ");
    //print(play);
    var bet = Math.ceil(cash/4);
    cash -= bet;
    ball = parseInt(play[0]);
    call = play[1];
    if (call == "PLAIN") {
        target = parseInt(play[2]);
    }

    x = call == "ODD";
    if (call == "PLAIN" && ball == target) {
        cash  += bet * 36;
    }else if (call == "EVEN" && ball > 0 && ball % 2 == 0) {
        cash += bet * 2;
    }else if (call == "ODD" && ball % 2 == 1) {
        cash += bet * 2;
    }
}

print(cash);